sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2553,60 +1914,19 @@ def stage_block(
         :returns: Blob property dict.
         :rtype: dict[str, Any]
         """
-        options = self._stage_block_options(
-            block_id,
-            data,
+        options = _stage_block_options(
+            require_encryption=self.require_encryption,
+            key_encryption_key=self.key_encryption_key,
```

Comment:
Move encryption check out, same as page and append blobs. Don't forget about doing this in async too.